syntax = "proto3";

package stereo_vehicle;

service ImageProcessing {
  // Find the height of a pixel
  rpc FindHeight(HeightRequest) returns (NumberResponse);
  
  // Identify vehicle in a video
  rpc IdentifyVehicle(IdentifyVehicleRequest) returns (BooleanArrayResponse);
}

// Request for height calculation
message HeightRequest {
  bytes left_image = 1;      // Left stereo image (binary data)
  bytes right_image = 2;     // Right stereo image (binary data)
  int32 x = 3;               // X coordinate
  int32 y = 4;               // Y coordinate
  float focal_length = 5;     // Focal length
  float baseline = 6;         // Baseline distance between cameras
  float camera_height = 7;    // Camera height from the ground
}

// Request for vehicle identification
message IdentifyVehicleRequest {
  bytes image = 1;           // Reference image for the vehicle
  bytes video = 2;           // Video containing vehicle (binary data)
}

// Response returning a single number
message NumberResponse {
  float number = 1;
}

// Response returning a boolean array
message BooleanArrayResponse {
  repeated bool results = 1;
}
